CMakeLists.txt

```diff
@@ -116,14 +116,68 @@ endif()
 
 # Add Rythe
 set(RYTHE_ENABLED_MODULE_NAMES "")
+set(RYTHE_ENABLED_APPLICATION_NAMES "")
 set(RYTHE_ENABLED_MODULE_PATHS "")
 add_subdirectory(${RYTHE_DIR_RYTHE})
 
 # The engine is ready - add applications
-if (${RYTHE_BUILD_APPLICATIONS})
+if ((${RYTHE_BUILD_APPLICATIONS}))
 	add_subdirectory(${RYTHE_DIR_ROOT}/applications)
 
 	foreach(path ${RYTHE_ENABLED_MODULE_PATHS})
 		add_subdirectory(${RYTHE_DIR_ROOT}/${path}/applications)
 	endforeach()
 endif()
+
+# Add config project
+set(configure_command_args "")
+list(APPEND configure_command_args "\"${RYTHE_DIR_ROOT}\"")
+list(APPEND configure_command_args "-G \"${CMAKE_GENERATOR}\"")
+list(APPEND configure_command_args "-S \"${RYTHE_DIR_ROOT}\"")
+list(APPEND configure_command_args "-B \"${CMAKE_BINARY_DIR}\"")
+list(APPEND configure_command_args -T ClangCL)
+list(APPEND configure_command_args -D RYTHE_FETCH_SUBMODULES=OFF)
+
```

Comment:
Can you not just do `list(APPEND configure_command_args -D RYTHE_BUILD_APPLICATIONS=${RYTHE_BUILD_APPLICATIONS})` ?

(and the same for ASAN, WERROR, ENABLE_ALL_MODULES) ?